pkg/service/HelmAppService.go

```diff
@@ -1016,11 +1038,63 @@ func (impl HelmAppServiceImpl) getHelmClient(clusterConfig *client.ClusterConfig
 		},
 		RestConfig: conf,
 	}
-
 	helmClientObj, err := helmClient.NewClientFromRestConf(opt)
 	if err != nil {
 		impl.logger.Errorw("Error in building client from rest config ", "err", err)
 		return nil, err
 	}
 	return helmClientObj, nil
 }
+
+func (impl HelmAppServiceImpl) HelmInstallCustom(request *client.HelmInstallCustomRequest) (bool, error) {
+	releaseIdentifier := request.ReleaseIdentifier
+	helmClientObj, err := impl.getHelmClient(releaseIdentifier.ClusterConfig, releaseIdentifier.ReleaseNamespace)
+	if err != nil {
+		impl.logger.Errorw("error on helm install custom", "err", err)
+		return false, err
+	}
+	var b bytes.Buffer
+	writer := gzip.NewWriter(&b)
+	_, err = writer.Write(request.Chunk.Content)
+	if err != nil {
+		impl.logger.Errorw("error on helm install custom", "err", err)
+		return false, err
+	}
+	err = writer.Close()
+	if err != nil {
+		impl.logger.Errorw("error on helm install custom", "err", err)
+		return false, err
+	}
+	err = os.MkdirAll(chartWorkingDirectory, os.ModePerm)
```

Comment:
os.MkdirAll(chartWorkingDirectory, os.ModePerm) should be call one time only (may be in NewHelmAppServiceImpl)

Reply:
done